[xinen26] iP

xinen26 opened the pull request:
# This my pull request!

## This is things I want to implement if time permits
- [x] massops
- [ ] update
- [ ] help

## This is my bullet list
* item x
* item y
* item z
## This is my numbered list 
1. item 1
2. item 2
3. item 3
## This is my fenced code block

```javascript
function abc() {
  console.log("The English Alphabet consists of 26 letters: A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z.");
}
```

## This is my task list 
- [ ] drink matcha latte :tea:
- [ ] eat lemon cookies :lemon:
- [ ] bake a bagel :bagel:

## This is my blockquote
> data engineer
> machine learning engineer 
> data architect

## This is my hyperlink 
This is my favourite site [DataCamp](https://www.datacamp.com/)

## This is my inline code
```
function abc() {
  console.log("The English Alphabet consists of 26 letters: A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z.");
}
```
## This is my text formatting
**This is bold text**
_This text is italicized_
~~This was mistaken text~~
This is a <sup>superscript</sup> text
This is a <sub>subscript</sub> text

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

xinen26 committed:
Level-0

xinen26 committed:
Level 1. Echo

xinen26 committed:
Level 2. Add, List

xinen26 committed:
Level 1. Echo

xinen26 committed:
Level 3. Mark as Done ROUGH

xinen26 committed:
Level 3. Mark as Done

xinen26 committed:
Level 4. ToDos, Events, Deadlines

xinen26 committed:
A-TextUiTesting: Automated Text UI Testing

xinen26 committed:
Level 5. Handle Errors

xinen26 committed:
Level 6. Delete

xinen26 committed:
Add storage functions

xinen26 committed:
Add basic functionalities of date

xinen26 committed:
Add printing function

xinen26 committed:
Add date related exceptions

xinen26 committed:
Handle exceptions for time

xinen26 committed:
Add Ui class

xinen26 committed:
Add each class to a package

xinen26 committed:
Merge branch 'add-gradle-support'

xinen26 committed:
Add gradle without errors

xinen26 committed:
Use JUnit to two non-trivial methods

xinen26 committed:
Change shadowjar stuff

xinen26 committed:
Add java doc comments for all classes

xinen26 committed:
Adhere to checkstyle coding standard

xinen26 committed:
Add find function

xinen26 committed:
Merge tag 'A-CodingStandard'

xinen26 committed:
Merge tag 'A-JavaDoc' into Level-9

xinen26 committed:
Merge tag 'Level-9' with coding standards

xinen26 committed:
Deal with all merge conflicts

xinen26 committed:
Merge branch 'master' into branch-Level-9

xinen26 committed:
Add small minor changes to code

Ko-Khan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,91 @@
+     * @return A string representing the status icon.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : "O"); // mark done task with X
```
Great way of keeping if-else statement short and sweet

Ko-Khan commented on src/main/java/duke/action/MyList.java:
```diff
@@ -0,0 +1,43 @@
+    /**
+     * The task list to be displayed.
+     */
+    private final TaskList mylist;
```
Wouldn't something like tasks be more appropriate  according to the coding standard?

Ko-Khan commented on src/main/java/duke/action/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+     * @param index The index of the task.
+     * @return The task at the specified index or null if the index is invalid.
+     */
+    public Task get(int index) {
```
Wouldn't getTask be a more appropriate name for this method?

Ko-Khan commented on src/main/java/duke/action/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+    /**
+     * Prints a goodbye message.
+     */
+    public void goodBye() {
```
Shouldn't method names be verbs?

Ko-Khan commented on src/main/java/duke/action/Mark.java:
```diff
@@ -0,0 +1,20 @@
+     * @return A string representing the response message.
+     */
+    @Override
+    public String response() {
```
Wouldn't a method name like getResponse fit the coding standard better?

Ko-Khan reviewed (commented):
I think some names don't follow naming conventions, but mostly great

Ko-Khan commented on src/main/java/duke/action/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+    /**
+     * The list containing tasks.
+     */
+    private final ArrayList<Task> mylist;
```
Shouldn't plural form be used for collection of objects?

Ko-Khan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.io.File;
```
I like the ordering of import statements. Maybe it would be better to have a space between java.io groups and others?

Ko-Khan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,137 @@
+                String status = token[1];
+                String description = token[2];
+                switch (taskType) {
+                case "T":
```
I like the identation for this switch statement

Ko-Khan commented on src/main/java/duke/action/Delete.java:
```diff
@@ -0,0 +1,39 @@
+     */
+    @Override
+    public String response() {
+        return "Noted. I've removed this task:\n" + deletedTask.toString() + "\nNow you have "
```
I like the splitting of longer lines throughout this pr.

Ko-Khan commented on src/main/java/duke/action/MyList.java:
```diff
@@ -0,0 +1,43 @@
+
+        for (int i = 0; i < mylist.size(); i++) {
+            Task task = mylist.get(i);
+            stringBuilder.append("  ").append(i + 1).append(". ").append(task).append("\n");
```
Maybe this line could have a line break?

Ko-Khan reviewed (commented):
I think you followed coding standards really well. Just a few nitpicks here and there

drustanyjt commented on data/tasks.txt:
Is the tracking of this saved file intentional? I personally chose to leave it out since I felt users would not want to make use of a copy of Duke with preloaded data.

drustanyjt commented on src/main/java/duke/CommandParser.java:
```diff
@@ -0,0 +1,150 @@
+                    String keyword = command.substring(5).trim();
+                    taskList.matches(keyword);
+                } else {
+                    System.out.println("Please provide a keyword to search for.");
```
In most other instances of commands being used, you throw some kind of exception if the input format is not of the intended form. Is there a reason that you print to the console in this case?

drustanyjt commented on src/main/java/duke/CommandParser.java:
```diff
@@ -0,0 +1,150 @@
+     * Parses user commands and performs corresponding actions in the Duke application.
+     */
+    public static Action parseCommand(String command, TaskList taskList) throws
+            EmptyDescriptionException, NoIndexException, UnknownCommandException,
+            WrongDateFormatException, WrongDateOrderingException {
```
Since these are all subclasses of `DukeException` why not use `throws DukeException` instead?

drustanyjt commented on src/main/java/duke/exception/DukeException.java:
I really like how you've split your Exceptions into multiple messages. I feel it makes very easy to predict what errors could occur.

drustanyjt commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,21 @@
+     */
+    @Override
+    public String getMessage() {
+        return "Unfortunately ";
```
It feels like you do not want a DukeException to be instantiated directly, because it's `getMessage` method would always just have the message `Unfortunately`. Maybe it would be a good idea to make it an abstract class, even if it has no abstract methods?

drustanyjt commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,91 @@
+     * @return A string representing the task type code.
+     */
+    public String getType() {
+        return "D";
```
Why does `Task` have a Type of `D`? Would it be more appropriate to make `Task` an abstract class, and make `getType` an abstract method?

xinen26 committed:
Fix minor bugs in gradle

xinen26 committed:
Update gradle

xinen26 committed:
Add javaFX tutorial one edits

xinen26 committed:
Add javaFX tutorial two edits

xinen26 committed:
Add javaFX tutorial 3 edits

xinen26 committed:
Add javaFX tutorial four edits

xinen26 committed:
Add a functioning GUI

xinen26 committed:
Edit implementation of Mark, Match, Unmark and add some exception

xinen26 committed:
Edit all the codes under Action interface

xinen26 committed:
Clean my code and add some javadocs

xinen26 committed:
Find command: make matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's,
* update the search algorithm to use case-insensitive matching
* add a script to migrate stress tests to the new format

xinen26 committed:
parseCommand command: make commands case-insensitive

parseCommand command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the command.

Let's,
* update the parseCommand to use case-insensitive matching

xinen26 committed:
Add assertion

xinen26 committed:
Clean code to adhere to code quality

xinen26 committed:
Merge pull request #1 from xinen26/branch-A-CodeQuality

Merge branch-A-CodeQuality to master branch

xinen26 committed:
Merge pull request #2 from xinen26/branch-A-Assertions

Merge branch-A-Assertions to master branch

xinen26 committed:
Add C-DetectDuplicates

xinen26 committed:
Add massops for delete, mark and unmark

xinen26 committed:
add welcome message

xinen26 committed:
Delete MyList file since its redundant

xinen26 committed:
Remove long method from command parser class

xinen26 committed:
Remove magic numbers

xinen26 committed:
Add more exception class

xinen26 committed:
Reduce code duplication

xinen26 committed:
reduce usage of long methods

xinen26 committed:
Improve on code quality by clearing checkstyle

xinen26 committed:
Add trivial help message

xinen26 committed:
Add trivial background color

xinen26 committed:
Improve readability of help message

xinen26 committed:
Change Duke to Maltese

xinen26 committed:
Edit code using feedback from soc-se-script

xinen26 committed:
Make file paths final static strings

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Managed to create the function to change file path. Only works when the file already exist though it will lead to null point error, the newfile will still get created but the tasks dont get imported. Yet to deal with exceptions as well. Tried to implement guide but could only work on command line. Will try to incorporate ChangeFilePath to point to sample.txt

xinen26 committed:
Works irregardless of whether its a new file or old file. Yet to deal with exceptions. Tried to incorporate ChangeFilePath to point to sample.txt but problem is the original sample data gets overwritten.

xinen26 committed:
Dealt with exceptions. Yet to deal with the overwritten issue. Considering other options such as manually adding. As for purge, maybe have a tracker to delete all data.

xinen26 committed:
Input my features into user guide

xinen26 committed:
Merge branch 'branch-C-Help'

* branch-C-Help:
   Created Guide and Help. However, Guide might not be actually implemented as it overwrites sample data

xinen26 committed:
Add exception for when the Deadline class has an invalid format

xinen26 committed:
Refactor CommandParser class

xinen26 committed:
Edit my tests for Parser class

xinen26 committed:
Adhere to coding standard

xinen26 committed:
Remove I/O redirection style automated testing

xinen26 committed:
Add more assertions

xinen26 committed:
Remove guide class as it is not functioning as expected

xinen26 committed:
Abort text-based UI

xinen26 committed:
Adhere to code quality

xinen26 committed:
Clear Checkstyle checks on all my files

xinen26 committed:
Update Ui.png

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md

xinen26 committed:
Update README.md
